Code/RDGeneral/Dict.cpp

```diff
@@ -45,12 +51,27 @@ bool Dict::getValIfPresent(const std::string &what, std::string &res) const {
   //  other casts, which will then be lexically cast to type T.
   //
   for(size_t i=0; i< _data.size(); ++i) {
-    if (_data[i].key == what) {
+    if (_data[i].getKey() == tag) {
       rdvalue_tostring(_data[i].val, res);
       return true;
     }
   }
   return false;
 }
 
+
+namespace common_properties {
+
+const char *getPropName(int v) {
+  if (v>=0 && v<=MAX) {
+    return common_properties::propnames[v];
+  }
+  if ((size_t)v < RDKit::KeyIntPair::tagmap.keys.size())
+    return RDKit::KeyIntPair::tagmap.keys[v].c_str();
```

Comment:
Are you suggesting something like this for storage?  This would remove relocation of the strings.

```
// key storage
std::vector<char *> keys;

//add key
const char * news= new char[sz];
memcpy(news, olds, sz); 

keys.push_back( news );
```